exporter/datadogexporter/config.go

```diff
@@ -490,6 +482,36 @@ func (c *Config) Validate() error {
 	return nil
 }
 
+func validateClientConfig(cfg confighttp.ClientConfig) error {
```

Comment:
A downside of this is that if new options are added to `confighttp.ClientConfig`, they won't be caught by this check. I guess using reflection would work here, but it's a bit of a pain

Reply:
I'd avoid using reflection too. In this case reflection also won't give us the mapstructure key, so we either can only return a generic error on new options or can only use the struct field name (which is different from mapstructure key), either seems ideal.